### PR TITLE
HDDS-9417. OM key delete use OzoneLockStrategy Acquisition Lock

### PR DESCRIPTION
**Pull request by guohao-rosicky**
## What changes were proposed in this pull request?

OM key delete use OzoneLockStrategy Acquisition Lock

The current use is to write locks to the bucket, this pr uses OzoneLockStrategy to use different locking strategies depending on the BucketLayout.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9417

## How was this patch tested?

Add support for OzoneLockStrategy in TestOMKeyRequest and test it with TestOMKeyDeleteRequest

### CODE REVIEW

**Comment by guohao-rosicky** [comment]
ci passed see: https://github.com/guohao-rosicky/ozone/actions/runs/6466797461
PTAL Thanks.  @duongkame @aswinshakil

**Comment by kerneltime** [comment]
@tanvipenumudy can you take a look as well?

**Comment by kerneltime** [comment]
@guohao-rosicky is this to improve the prefix locking? Right now it is disabled by default.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyRequest.java [hunk 1]
```diff
@@ -234,6 +239,19 @@ public void setup() throws Exception {
     when(ozoneManager.getOmSnapshotManager())
         .thenReturn(omSnapshotManager);
 
+
+    final boolean keyPathLockEnabled =
+        ozoneConfiguration.getBoolean(OZONE_OM_ENABLE_FILESYSTEM_PATHS,
+            OZONE_OM_ENABLE_FILESYSTEM_PATHS_DEFAULT);
+    final boolean fileSystemPathsEnable =
```
Comment (tanvipenumudy) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyRequest.java:246]: Can we parameterize the tests based on the configuration flags passed: `ozone.om.key.path.lock.enabled` and `ozone.om.enable.filesystem.paths` to either use `OBSKeyPathLockStrategy` (finer `KEY_PREFIX_LOCK`) or `RegularBucketLockStrategy` (`BUCKET_LOCK`) pattern implementations depending on this [logic](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneLockProvider.java#L37-L52)?

**Review by tanvipenumudy — COMMENTED** [review]
Currently, `OzoneLockStrategy` patterns are only being used for key create request. 

Is the motivation behind this change to reduce lock contention by changing behavior of the locking mechanism (when `ozone.om.key.path.lock.enabled` is set to `true`) at runtime?

**Comment by guohao-rosicky** [comment]
> Currently, `OzoneLockStrategy` patterns are only being used for key create request.
> 
> Is the motivation behind this change to reduce lock contention by changing behavior of the locking mechanism (when `ozone.om.key.path.lock.enabled` is set to `true`) at runtime?

Actually this PR I just want to improve the speed of deletion of a bucket with a layout like object store.

For other layout buckets I agree with what you said about

```
Can we parameterize the tests based on the configuration flags passed: ozone.om.key.path.lock.enabled and ozone.om.enable.filesystem.paths to either use OBSKeyPathLockStrategy (finer KEY_PREFIX_LOCK) or RegularBucketLockStrategy (BUCKET_LOCK) pattern implementations depending on this [logic](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneLockProvider.java#L37-L52)

```

**Comment by adoroszlai** [comment]
/pending I'll try to update the UT

**Pull request closed without merging**